test/linter/test-status.js

```diff
@@ -0,0 +1,57 @@
+const chalk = require('chalk');
+const { Logger } = require('./utils.js');
+
+/**
+ * @typedef {import('../../types').Identifier} Identifier
+ */
+
+/**
+ * @param {Identifier} data
+ * @param {Logger} logger
+ * @param {string} path
+ */
+function checkStatus(data, logger, path = '') {
+  const compat = data.__compat;
+  if (compat?.status) {
+    if (compat.spec_url && compat.status.standard_track === false) {
+      logger.error(
+        chalk`{red → {bold ${path}} is marked as {bold non-standard}, but has a {bold spec_url}}`,
+      );
+    }
+    if (compat.status.experimental && compat.status.deprecated) {
+      logger.error(
+        chalk`{red → Unexpected simultaneous experimental and deprecated status in {bold ${path}}}`,
+      );
+    }
+  }
+
+  // Check children
+  for (const member in data) {
```

Comment:
Something with `Object.entries(data)` would make sense here since we need both key and value.

Although, can we just use utils.walk to visit all the entries?

Reply:
For this and the above comment, I plan to do as such in the next-gen linter!